src/main/java/com/hivemq/client/internal/mqtt/handler/subscribe/MqttSubscriptionHandler.java

```diff
@@ -93,142 +99,119 @@
     public void onSessionStartOrResume(
             final @NotNull MqttClientConnectionConfig connectionConfig, final @NotNull EventLoop eventLoop) {
 
-        super.onSessionStartOrResume(connectionConfig, eventLoop);
-        if (connectionConfig.areSubscriptionIdentifiersAvailable() && (subscriptionIdentifiers == null)) {
-            subscriptionIdentifiers = new Ranges(1, MqttVariableByteInteger.FOUR_BYTES_MAX_VALUE);
+        subscriptionIdentifiersAvailable = connectionConfig.areSubscriptionIdentifiersAvailable();
+
+        if (!hasSession) {
+            incomingPublishFlows.getSubscriptions().forEach((subscriptionIdentifier, subscriptions) -> {
```

Comment:
I think this bahavior should be configurable somehow.

Reply:
What should be configurable?
If you use automatic reconnect and the server lost the session between the connection loss and a successful reconnect, why would you not want to resubscribe?

Reply:
This is actually feedback from users who are confused by SessionExpiredExceptions when they use automatic reconnect.

Reply:
Use case: You are in extremely constrained networks and want to be very precise on which data you send and when you send it.

Reply:
The same is true for resending publishes.
Isn't automatic reconnect the wrong choice if you want to have control over everything?

Reply:
> The same is true for resending publishes.
I did not know that. But I would argue that the same logic applies.

I feel like the connect is something else. As it is the basis for everything afterwards.

Reply:
So would a config to disable automatic resubscribe be enough?

Reply:
Yeah for a start I think so.

Reply:
Added configuration option: `Reconnector.resubscribeIfSessionExpired` (default true)
Also added a similar option for republishing: `Reconnector.republishIfSessionExpired` (default false)